src/sdk.js

```diff
@@ -1,57 +1,111 @@
 window.Smartcar = (function(window) {
   'use strict';
+  // accesss denied error class
+  class AccessDenied extends Error { // eslint-disable-line no-restricted-syntax
+    constructor(message) {
+      super(message);
+      this.name = 'AccessDenied';
+    }
+  }
 
   /**
    * Initializes Smartcar class
    *
    * @param {Object} options - the SDK configuration object
    * @param {String} options.clientId - the applications' client id
-   * @param {String} options.redirectUri - the registered redirect uri of the application
+   * @param {String} options.redirectUri - the registered redirect uri of the
+   * application
    * @param {String[]} [options.scope] - requested permission scopes
-   * @param {Function} [options.onComplete] - called upon completion of the Auth flow
-   * @param {Boolean} [options.development=false] - launch Smartcar auth in development mode
-   * to enable the mock vehicle brand
+   * @param {Function} [options.onComplete] - called on completion of auth flow
+   * @param {Boolean} [options.development=false] - launch Smartcar auth in
+   * development mode to enable the mock vehicle brand
    * @constructor
    */
   function Smartcar(options) {
+    // allow only one instance of Smartcar
+    if (Smartcar._hasBeenInstantiated) {
+      throw new Error('Smartcar has already been instantiated in the window.' +
+        ' Only one instance of Smartcar can be defined. See' +
+        ' https://github.com/smartcar/javascript-sdk for more information');
+    } else {
+      Smartcar._hasBeenInstantiated = true;
+    }
+
+    if (!options.redirectUri) {
+      throw new TypeError('A redirect URI option must be provided');
+    }
+
+    if (!options.clientId) {
+      throw new TypeError('A client ID option must be provided');
+    }
+
+    // require onComplete method with at least two parameters (error & code)
+    // when hosting on Smartcar CDN
+    if (options.redirectUri.startsWith('https://cdn.smartcar.com')
```

Comment:
same here on the url

Reply:
also do we validate here that the url has an "app_origin" query param?

Reply:
that sounds like a fantastic idea, will add